app/models/address.rb

```diff
@@ -0,0 +1,29 @@
+class Address < ActiveRecord::Base
+  validates :street_1, presence: true
+
+  validates(
+    :latitude,
+    presence: true,
+    numericality: { greater_than_or_equal_to: -90, less_than_or_equal_to: 90 }
+  )
+
+  validates(
+    :longitude,
+    presence: true,
+    numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
+  )
+
+  validates :state_code, presence: true
+
+  validates_presence_of(
+    :city,
+    unless: proc { |a| a.zip_code.present? },
+    message: "can't be empty if zip code is empty"
+  )
+
+  validates_presence_of(
```

Comment:
Any particular reason for switching away from [sexy validations](https://github.com/bbatsov/rails-style-guide#sexy-validations)?